tests/test_main.py

```diff
@@ -2251,33 +2252,50 @@ class Model(BaseModel):
 
 
 def test_protected_namespace_default():
-    with pytest.raises(NameError, match='Field "model_prefixed_field" has conflict with protected namespace "model_"'):
+    warnings.filterwarnings('error')
+    try:
+        with pytest.raises(
```

Comment:
Maybe just `pytest.warns`? (also below)

Reply:
yep, no idea why I did  this here. I did need to do it in the docs because I wanted to print / not use pytest in the doc tests, but here, pytest.warns is definitely better.